diesel_derives/src/lib.rs

```diff
@@ -880,10 +882,12 @@ pub fn derive_queryable_by_name(input: TokenStream) -> TokenStream {
 ///   the entire select expression for the given field. It may be used to select with
 ///   custom tuples, or specify `select_expression = my_table::some_field.is_not_null()`,
 ///   or separate tables...
-///   It should be used in conjunction with `select_expression_type` (described below)
-/// * `#[diesel(select_expression_type = the_custom_select_expression_type]`, to be used
-///   in conjunction with `select_expression` (described above).
-///   For example: `#[diesel(select_expression_type = dsl::IsNotNull<my_table::some_field>)]`
+///   It may be used in conjunction with `select_expression_type` (described below)
+/// * `#[diesel(select_expression_type = the_custom_select_expression_type]`, should be used
```

Comment:
Maybe list cases that continue to require this? (I would assume that if `select_expression` is a custom function call it might be required?)